app/uk/gov/hmrc/individualsmatchingapi/controllers/v2/PrivilegedCitizenMatchingController.scala

```diff
@@ -18,33 +18,42 @@ package uk.gov.hmrc.individualsmatchingapi.controllers.v2
 
 import javax.inject.{Inject, Singleton}
 import org.slf4j.LoggerFactory
+import play.api.hal.Hal.links
+import play.api.hal.HalLink
 import play.api.libs.json.Json
+import play.api.mvc.hal._
 import play.api.mvc.{BodyParsers, ControllerComponents}
 import uk.gov.hmrc.auth.core._
+import uk.gov.hmrc.individualsmatchingapi.domain.JsonFormatters._
 import uk.gov.hmrc.individualsmatchingapi.controllers.Environment._
 import uk.gov.hmrc.individualsmatchingapi.controllers.{CommonController, PrivilegedAuthentication}
+import uk.gov.hmrc.individualsmatchingapi.domain.CitizenMatchingRequest
 import uk.gov.hmrc.individualsmatchingapi.services.{CitizenMatchingService, LiveCitizenMatchingService, SandboxCitizenMatchingService, ScopesService}
 
-import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.ExecutionContext
 
 abstract class PrivilegedCitizenMatchingController(
   citizenMatchingService: CitizenMatchingService,
   scopeService: ScopesService,
-  cc: ControllerComponents)
+  cc: ControllerComponents)(implicit val ec: ExecutionContext)
     extends CommonController(cc) with PrivilegedAuthentication {
 
-  private def seeOthers(location: String) =
-    new Status(SEE_OTHER)(Json.obj()).withHeaders(LOCATION -> location)
-
   val logger = LoggerFactory.getLogger(this.getClass)
 
   def matchCitizen = Action.async(BodyParsers.parse.json) { implicit request =>
-    val scopes = scopeService.getAllScopes
-    requiresPrivilegedAuthentication(scopes)
-      .flatMap { authScopes =>
-        throw new Exception("NOT_IMPLEMENTED")
-      }
-      .recover(recovery)
+    requiresPrivilegedAuthentication(scopeService.getAllScopes) { _ =>
+      withJsonBody[CitizenMatchingRequest] { matchCitizen =>
+        citizenMatchingService.matchCitizen(matchCitizen) map { matchId =>
+          val selfLink = HalLink("self", s"/individuals/matching/")
+          val individualLink = HalLink(
+            "individual",
+            s"/individuals/matching/$matchId",
+            name = Option("GET"),
+            title = Option("Individual Details"))
```

Comment:
Could this cause confusion with the Individuals Details API? Do we need to rethink this title? Maybe one for Lisa?